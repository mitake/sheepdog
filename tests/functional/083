#!/bin/bash

# Test vdi repair functionality for corrupted inodes

. ./common

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format

$DOG vdi create test 12M
_vdi_list

echo "original data" | $DOG vdi write test

$DOG cluster shutdown

# single inode object corrupted

_random | dd of=$STORE/0/obj/807c2b2500000000 bs=4096 count=2048 &> /dev/null
for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

echo vdi test should be visible

_vdi_list

$DOG vdi read test 0 14

$DOG cluster shutdown

