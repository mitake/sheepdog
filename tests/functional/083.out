QA output created by 083
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   12 MB  0.0 MB  0.0 MB DATE   7c2b25     3              
vdi test should be visible
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   12 MB  4.0 MB  0.0 MB DATE   7c2b25     3              
original data
